app/parse-siti-payment-file/parse-filename.js

```diff
@@ -0,0 +1,18 @@
+const mapData = (data, maskData) => {
+  let i = 0
+  Object.keys(maskData).forEach((key) => {
+    maskData[key] = data[i++]
+  })
+  return maskData
+}
+
+const parseFilename = (filename, filenameMask) => {
+  filename = filename.replace(/_/g, '')
+  let i = 0
+  const convertToMask = filenameMask.mask.replace(/#/g, () => {
```

Comment:
As it's a single statement you can simplify:

`filenameMask.mask.replace(/#/g, () => filename[i++])`